packages/react/src/hooks/useResizeObserver.ts

```diff
@@ -59,5 +59,5 @@ export function useResizeObserver<T extends HTMLElement>(
     }
 
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [target, ...depsArray])
+  }, [target?.current, ...depsArray])
```

Comment:
I know you originally moved line 25 outside of the hook and put the dependency on the `targetEl` but that was generating SSR issues due to document not being defined, was wondering how you felt about this alternative instead `@siddharthkp`